src/lib/Name.hs

```diff
@@ -3326,6 +3332,78 @@ toListNameMap :: NameMap c a n -> [(Name c n, a)]
 toListNameMap (UnsafeNameMap raw) = R.toList raw <&> \(r, x) -> (UnsafeMakeName r, x)
 {-# INLINE toListNameMap #-}
 
+unionWithNameMap :: (a -> a -> a) -> NameMap c a n -> NameMap c a n -> NameMap c a n
+unionWithNameMap f (UnsafeNameMap raw1) (UnsafeNameMap raw2) =
+  UnsafeNameMap $ R.unionWith f raw1 raw2
+{-# INLINE unionWithNameMap #-}
+
+traverseNameMap :: (Applicative f) => (a -> f b)
+                 -> NameMap c a n -> f (NameMap c b n)
+traverseNameMap f (UnsafeNameMap raw) = UnsafeNameMap <$> traverse f raw
+{-# INLINE traverseNameMap #-}
+
+mapNameMap :: (a -> b) -> NameMap c a n -> (NameMap c b n)
+mapNameMap f (UnsafeNameMap raw) = UnsafeNameMap $ fmap f raw
+{-# INLINE mapNameMap #-}
+
+newtype NameMapE (c::C) (e:: E) (n::S) = NameMapE (NameMap c (e n) n)
+  deriving (Eq, Semigroup, Monoid)
+
+hoistFilterNameMapE :: (BindsNames b, HoistableE e, ShowE e)
+                    => b n l -> NameMapE c e l -> HoistExcept (NameMapE c e n)
+hoistFilterNameMapE b (NameMapE nmap) =
+  NameMapE <$> (traverseNameMap (hoist b) $ hoistFilterNameMap b nmap) where
+{-# INLINE hoistFilterNameMapE #-}
+
+hoistFilterNameMapEWith :: (BindsNames b, HoistableE e, ShowE e)
+  => b n l -> (Abs b e n -> e n) -> NameMapE c e l -> NameMapE c e n
+hoistFilterNameMapEWith b f (NameMapE nmap) =
+  NameMapE $ mapNameMap doit $ hoistFilterNameMap b nmap where
+  doit v = f $ Abs b v
+{-# INLINE hoistFilterNameMapEWith #-}
+
+hoistFilterNameMapEWithM
+  :: (Applicative f, BindsNames b, HoistableE e, ShowE e)
+  => b n l -> (Abs b e n -> f (e n)) -> NameMapE c e l -> f (NameMapE c e n)
+hoistFilterNameMapEWithM b f (NameMapE nmap) =
```

Comment:
What does this method do? It's a little hard to follow.

Reply:
This isn't needed anymore, so I'll just delete it.